clients/src/test/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinatorTest.java

```diff
@@ -176,6 +176,41 @@ public void testJoinGroupRequestMaxTimeout() {
         assertTrue(consumerClient.poll(future, mockTime.timer(0)));
     }
 
+    @Test
+    public void testJoinGroupRequestWithMemberIdRequired() {
+        setupCoordinator();
+        mockClient.prepareResponse(groupCoordinatorResponse(node, Errors.NONE));
+        coordinator.ensureCoordinatorReady(mockTime.timer(0));
+
+        final String memberId = "memberId";
+        final int generation = 10;
+
+        mockClient.prepareResponse(joinGroupFollowerResponse(generation, memberId, "leaderId", Errors.MEMBER_ID_REQUIRED));
```

Comment:
I'm not sure it makes sense to use a valid generation and leaderId here. The coordinator actually returns generationId=0 and leaderId="" at the moment for the MEMBER_ID_REQUIRED case.

As an aside, we should probably fix `GroupCoordinator.joinError()`. I can't think of a good reason why we should return 0 instead of `GroupCoordinator.NoGeneration`. Can we do that in this patch?

Reply:
Yep, fixing it.